### PR TITLE
docs(typing): snippets split, improved, and lint

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #1453 

### Changes

Changes snippet fictitious names to real code snippet filenames for easier discovery and collaboration to improve/correct document snippets.

Some examples have also been refactored so that the codes are complete and there are no lint/pre-commit hooks issues.

### User experience

Before:
image

After:
image
image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

-----
[View rendered docs/utilities/typing.md](https://github.com/leandrodamascena/aws-lambda-powertools-python/blob/chore/typing/docs/utilities/typing.md)

### CODE REVIEW

File: docs/utilities/typing.md [hunk 1]
```diff
@@ -7,17 +7,40 @@ description: Utility
 
 This typing utility provides static typing classes that can be used to ease the development by providing the IDE type hints.
 
-![Utilities Typing](../media/utilities_typing.png)
+## Key features
+
+* Add static typing classes
+* Ease the development by providing the IDE type hint
```
Comment (rubenfonseca) [inline docs/utilities/typing.md]:
```suggestion
* Ease the development by leveraging your IDE's type hints
```

File: docs/utilities/typing.md [hunk 2]
```diff
@@ -7,17 +7,40 @@ description: Utility
...
-from aws_lambda_powertools.utilities.typing import LambdaContext
+=== "getting_started_validator_decorator_function.py"
+
+	```python hl_lines="1 4"
+    --8<-- "examples/typing/src/getting_started_typing_function.py"
+	```
+
+## Working with context methods and properties
+
+Using `LambdaContext` typing it is possible to access information and hints of all properties and methods implemented by Lambda context object.
```
Comment (rubenfonseca) [inline docs/utilities/typing.md]:
```suggestion
Using `LambdaContext` typing makes it possible to access information and hints of all properties and methods implemented by Lambda context object.
```

**Review by rubenfonseca — CHANGES_REQUESTED** [review]
Some small things, just suggestions, feel free to re-write :)

File: docs/utilities/typing.md (hunk 1 above)
Comment (leandrodamascena) [reply docs/utilities/typing.md]: Yup, this is better than my text!

File: docs/utilities/typing.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ The `LambdaContext` typing is typically used in the handler method for the Lambd
 
 ## Working with context methods and properties
 
-Using `LambdaContext` typing it is possible to access information and hints of all properties and methods implemented by Lambda context object.
+Using `LambdaContext` typing make it possible to access information and hints of all properties and methods implemented by Lambda context object.
```
Comment (rubenfonseca) [inline docs/utilities/typing.md]:
```suggestion
Using `LambdaContext` typing makes it possible to access information and hints of all properties and methods implemented by Lambda context object.
```

**Review by rubenfonseca — APPROVED** [review]
Just a small thing that you can approve and LGTM

File: examples/typing/src/working_with_context_function.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+def lambda_handler(event, context: LambdaContext) -> dict:
+
+    limit_execution: int = 1000  # milliseconds
+
+    # scrape website and exit before lambda timeout
+    while context.get_remaining_time_in_millis() > limit_execution:
+
+        comments: requests.Response = requests.get("https://jsonplaceholder.typicode.com/comments")
+        # add logic here and save the results of the request to an S3 bucket, for example.
+        ...
```
Comment (heitorlessa) [inline examples/typing/src/working_with_context_function.py]:
```suggestion

```

**Pull request merged**